doc/visual-programming/source/widgets/model/stochasticgradient.rst

```diff
@@ -0,0 +1,103 @@
+Stochastic Gradient Descent
+===========================
+
+.. figure:: icons/stochastic-gradient.png
+
+Minimize an objective function using a stochastic approximation of gradient descent.
+
+Signals
+-------
+
+**Inputs**:
+
+-  **Data**
+
+   A data set.
+
+-  **Preprocessor**
+
+   Preprocessing method(s)
+
+**Outputs**:
+
+-  **Learner**
+
+   A SGD regression learning algorithm with settings as specified in the dialog.
+
+-  **Model**
+
+   A trained model. Output signal sent only if input *Data* is present.
+
+Description
+-----------
+
+The **Stochastic Gradient Descent** widget uses `stochastic gradient descent <https://en.wikipedia.org/wiki/Stochastic_gradient_descent>`_ that minimizes a chosen loss function with a linear function. The algorithm approximates a true gradient by considering one sample at a time, and simultaneously updates the model based on the gradient of the loss function. For regression, it returns predictors as minimizers of the sum, i.e. M-estimators, and is especially useful for large-scale and sparse data sets.
+
+.. figure:: images/StochasticGradientDescent-stamped.png
+   :scale: 50 %
+
+1. Specify the name of the model. The default name is "SGD".
+2. Algorithm parameters. Classification loss function:
+
+   -  `Hinge <>`_ ()
```

Comment:
@pavlin-policar Could you please check Stochastic Gradient (and SVM, too), since you were our chief 'merger'. These are the new additions that were not found in the old documentation. You can either add to this PR or just write the missing things in the comment.